src/NServiceBus.Core/MessageInterfaces/MessageMapper/Reflection/MessageMapper.cs

```diff
@@ -242,5 +245,7 @@ public object CreateInstance(Type t)
         ConcurrentDictionary<RuntimeTypeHandle, RuntimeTypeHandle> concreteToInterfaceTypeMapping = new ConcurrentDictionary<RuntimeTypeHandle, RuntimeTypeHandle>();
         ConcurrentDictionary<string, RuntimeTypeHandle> nameToType = new ConcurrentDictionary<string, RuntimeTypeHandle>();
         ConcurrentDictionary<RuntimeTypeHandle, RuntimeMethodHandle> typeToConstructor = new ConcurrentDictionary<RuntimeTypeHandle, RuntimeMethodHandle>();
+
+        private static readonly object messageInitializationLock = new object();
```

Comment:
I understand right now based on the description of the PR why you chose to make this static. The reason is because in the testing infra we might have multiple instance of the message mapper. Since this class is intended to be used as a singleton instance I propose the following:

* Switch to non-static lock object.
* Document on the class that this class is intended to be used as a singleton.
* Change the testing framework to make the mapper a singleton/static

How does that sound?